### PR TITLE
8.0 Add module l10n_fr_mis_reports

### PR DESCRIPTION
**Pull request by alexis-via**
This PR depends on https://github.com/OCA/account-financial-reporting/pull/86

For the moment, it is still a "work in progress", because it only contains P&L, not the balance sheet. But, before implementing the balance sheet, I would like to have some feedback on my implementation of the P&L in mis_reports: for P&L, I took the structure of the liasse fiscale (regular format, not the simplified format), but I don't know if it's the most appropriate... The different options available are:
- keep the structure of the liasse fiscale (regular format)
- if you think it's too detailed, switch to the structure of the liasse fiscale in simplified format (simplified format may be better for the MIS reports, which mainly aims at "dashboard" usage)
- take another structure... but which one would you propose ?

Another question is: is there a reference document on how to bind the accounts of the PCG to the P&L ? I found very few info about this on the Net ; the only relevant document I found was this one:
http://www.insee.fr/fr/ppp/bases-de-donnees/irweb/images05/dd/doc/pdf/images05_cptr.pdf
but it doesn't give a lot of details...

### CODE REVIEW

**Comment by sbidoul** [comment]
@alexis-via you can create several templates, no need to choose between the full and the simple one. The PDF export works quite well too, so it's not only for dashboards.

You can also use mis_builder as a computation engine (see aep.py). For instance in Luxembourg, we are considering rewriting the XML export of the equivalent of the Liasse Fiscale by using the mis_builder KPI definitions mapped to XML templates and use the mis_builder AEP to compute the values.

**Comment by alexis-via** [comment]
@sbidoul
You are right ; I could create several templates... one for the "standard" P&L/Balance sheet, and one for the simplified version.

I'll try to find time to finish the standard version soon.

**Comment by alexis-via** [comment]
@sbidoul 
I have a question on mis_builder:
On the Balance Sheet, I need to have the amount of the bank account as an asset if the amount in the bank is > 0, but I want to have the amount of the bank account as a debt if the amount in the bank is negative. How can I do that with mis_builder ? I didn't find an example of that in l10n_be_mis_reports (maybe it's different in BE reports...).

**Comment by alexis-via** [comment]
By the way, I asked an accountant where to find to binding the accounts of the PCG to the P&L/BS ? He told me I would find this information in the Lefevre Comptable. We have this book at the office, I'll have a look next time I'll be at the office.

File: .travis.yml
Comment (sbidoul) [inline .travis.yml]: you could take the opportunity to replace this by `oca_dependencies.txt`

File: l10n_fr_mis_reports/README.rst
Comment (sbidoul) [inline l10n_fr_mis_reports/README.rst]: Add `=========` above to make this a proper title

File: l10n_fr_mis_reports/README.rst
Comment (sbidoul) [inline l10n_fr_mis_reports/README.rst]: In the lastest (moving target :wink: README.rst template) I think there is a link target.
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: l10n_fr_mis_reports/README.rst
Comment (sbidoul) [inline l10n_fr_mis_reports/README.rst:54]: Add the bug tracker section.

Comment (sbidoul) [inline l10n_fr_mis_reports/README.rst:54]: You should also add the "test me on runbot button".
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

**Comment by alexis-via** [comment]
I confirm that the information is available in the Lefevre Comptable book.

File: .travis.yml
Comment (legalsylvain) [inline .travis.yml]: I don't understand why runbot is failing

> except_orm: ('Error', u"You try to install module 'l10n_fr_mis_reports' that depends on module 'mis_builder'.\nBut the latter module is not available in your system.").

Maybe use oca_dependencies will fix the bug.

**Comment by sbidoul** [comment]
I added OCA/account-financial-reporting and OCA/reporting-engine in the runbot dependencies and restarted the build.

File: .travis.yml
Comment (sbidoul) [inline .travis.yml]: @legalsylvain runbot is now green!

**Comment by MOSAMAD** [comment]
:+1:

**Comment by brother-bernard** [comment]
:+1:

**Comment by legalsylvain** [comment]
please wait a time before merging.

**Comment by gurneyalex** [comment]
needs rebasing

**Comment by alexis-via** [comment]
rebase done.

**Comment by gurneyalex** [comment]
travis build of 8.0 was recently greenified. can you rebase?

**Comment by alexis-via** [comment]
Due to the strong disagreement that took place in https://github.com/OCA/l10n-france/pull/63 and the lack of respect for my work in OCA/l10n-france, I decided to move this module l10n_fr_mis_reports to a new repository outside of the OCA. I will decide in the coming days the name and location of the new repository and I will announce it here.

**Comment by legalsylvain** [comment]
I talked with @alexis-via, I reopen this PR and we'll test it on Akretion instance in a while.

I added again 'need review' label, because I really think that this PR need external review. (As said previously, OCA PRs need neutral reviews). That is not the case for @MOSAMAD and @brother-bernard.

Regards.

**Comment by bealdav** [comment]
Please rename module in singular form as guidelines say
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

We claim consistency (me first) for others (ie odoo), don't propagate exceptions. Thanks

**Comment by alexis-via** [comment]
In OCA/l10n-belgium, the module is named "l10n_be_mis_reports".
In OCA/l10n-luxemburg, the module is named "l10n_lu_mis_reports".
That's why I named this module l10n_fr_mis_reports.

**Comment by bealdav** [comment]
Ok I understand. But when I see somebody drive on the bad side of the motorway, I try to stay on the right way (except if he is in front of me at a moment ;-) )
I'm sure now you'll notice to others to stay on the right way now, thanks a lot.

**Comment by gurneyalex** [comment]
:+1:

**Comment by sbidoul** [comment]
Sincere apologies for the trailing s.

:+1:

**Comment by bealdav** [comment]
;-)
:+1:

**Comment by alexis-via** [comment]
FYI, I still have a few issues to fix in balance sheet (P&L is OK AFAIK) and then I'll merge.

**Review by gurneyalex — APPROVED** [review]
changing my old way approval to a new mode one.

**Comment by gurneyalex** [comment]
needs rebasing

**Comment by sebastienbeau** [comment]
Build is red as 8.0 branch is red. AS code have been approved and work correctly I merge it

**Pull request merged**
